### ISSUE TITLE
myWebView.loadURL not access to file on Android

### ISSUE THREAD

**Issue by giucer** [airsdk/Adobe-Runtime-Support#1006]
When I pass to myWebView this:

ImgArtFile = File.applicationStorageDirectory.resolvePath("Images/" + data.toString() + ".jpg");
ImgArtSource = ImgArtFile.nativePath;
myWebView.loadURL(ImgArtSource);

on Windows and iOS works well....but when I try to rin on android this is the result:

![WhatsApp Image 2021-07-25 at 21 02 36](https://user-images.githubusercontent.com/56581167/126910392-3b244c70-84fc-4cd4-a07f-a8ed519e7aa9.jpeg)

**Comment by balaedit** [comment]
While using WebView, I was using cacheDirectory path, but the same result. Access Denied.

**Comment by marcanw** [comment]
Same here.
I can't make any new update and release because of this.
Will it be solved soon?

**Comment by ajwfrost** [comment]
We're looking at the storage in Android at the moment and trying to see how we can fit how Android want you to do everything (via their Storage APIs which imply user interaction) vs how AIR was set up to do things (with an expectation of synchronous file system access). So I hope we will be able to provide an update here soon..

thanks

**Comment by marcanw** [comment]
Thanks.

My main concern is to be able to have the access from a local html content as described here :
![image](https://user-images.githubusercontent.com/21979056/141854338-2b0460f4-f2a1-4408-8f87-e3b899b93dc0.png)
https://help.adobe.com/en_US/as3/dev/WS901d38e593cd1bac3ef1d28412ac57b094b-8000.html#WS901d38e593cd1bac-5a0b8b512ac674f48f-7fff

**Comment by giucer** [comment]
I've found a workaround but it need to know wich version of sdk is in use through a ANE library....this is my workaround:

					ImgArtFile = Click.ImageDir.resolvePath("Images/" + data.toString() + ".jpg");
					if (ImgArtFile.exists) {
							if ((Number(Click.AndroidSDK) > 0) && (Number(Click.AndroidSDK) < 29))
								ImgArtSource = ImgArtFile.url;
							else
								ImgArtSource = ImgArtFile.nativePath;
                                       }

**Comment by marcanw** [comment]
Your workaround doesn't solve my problem, at least for html documents.

On my side this my code:
```
var file:File = File.applicationStorageDirectory.resolvePath("home.html");		
var fileStream:FileStream = new FileStream();
fileStream.open(file, FileMode.WRITE);
fileStream.writeUTFBytes(st);
fileStream.close();
var url = file.nativePath;
if(url.indexOf("file://") != 0) url = "file://" + url;
homeView.load(url);
```
This code works till I set the android level to 30.
So maybe it's not the same problem. If so, I will create a separate issue.

**Comment by marcanw** [comment]
I found this https://developer.android.com/reference/android/webkit/WebSettings#setAllowFileAccess(boolean)

**Comment by giucer** [comment]
I'll explain better, with your example you have to write this code:

var file:File;
if (Number(Click.AndroidSDK) > 29)
	file = File.applicationStorageDirectory.resolvePath("home.html");
else
	file = File.documentsDirectory.resolvePath("home.html");		
var fileStream:FileStream = new FileStream();
fileStream.open(file, FileMode.WRITE);
fileStream.writeUTFBytes(st);
fileStream.close();
if (file.exists) {
	if ((Number(Click.AndroidSDK) > 0) && (Number(Click.AndroidSDK) < 29))
		homeView.load(file.url);
	else
		homeView.load(file.nativePath);
}

The value of the variable AndroidSDK is the result of a ANE library call that return this android value: Build.VERSION.SDK_INT

**Comment by ajwfrost** [comment]
@marcanw are you specifically looking to load in content from the application storage directory? i.e. arbitrary content? Or would you just want to load things in from the application directory itself i.e. files that would be included as assets in your APK?

The change - per the link you shared - seems to be an Android security update so it would still allow you to load a file that you had in your APK via file:///android_asset/filename.html, but it would block other file: URLs (unless you explicitly set this `setAllowFileAccess()` method to 'true').

If you really need to do this with arbitrary file URLs, then we can perhaps have an opt-in for that ... but I would expect Android to drop this function in the future and eliminate this capability, as they have for the `setAllowFileAccessFromFileURLs` function etc.

**Comment by marcanw** [comment]
@ajwfrost I need to load html from the application directory OR generated by the app. Inside the html I would like to dynamically load content generated by the app (file://...), content on our server (jpg, png, xml, json, ..) and call our server (php).

**Comment by marcanw** [comment]
@ajwfrost I already tried loadString but as you can see on the table above it doesn't give access to local resources nor to remote request.
For loadData, I use the load("data:...") but for big files like videos, it is too slow.

Sure for webViewAllowFileAccess if it can solve my problem. Any idea about a ETA?

Thank you for your help

**Comment by ajwfrost** [comment]
@marcanw good point "for big files like videos" if those are being loaded then that's a bit of a pain.

We're planning some updates which we would likely load up as a beta version in a couple of weeks so we can put this into that, it should be fairly straightforward...

thanks

**Comment by marcanw** [comment]
@ajwfrost Any news about this. I can't make any new update on Play Store for more than one month...
Thanks

**Comment by ajwfrost** [comment]
Sorry @marcanw I should have gone through and updated the tickets ... the beta is up there at the moment, and it should work now if you add that `webViewAllowFileAccess` tag per the above.

See https://airsdk.harman.com/download/33.1.1.698

thanks

**Comment by marcanw** [comment]
Hi @ajwfrost 
I tried it but I can't publish from Animate
I get this message.
![image](https://user-images.githubusercontent.com/21979056/145302510-1bb30bc2-5f16-41c6-946c-6e24e9ad261d.png)
What can I do?
thanks

**Comment by marcanw** [comment]
Hi @ajwfrost any news about the publishing issue with the air 3.1.1.698 on android?

**Comment by ajwfrost** [comment]
Hi @marcanw  - you need to set up an Android platform SDK, which can be added in the "[AIRSDK]\lib\adt.cfg" file (note you'll need to use \\ or / symbols on Windows, i.e. rather than "c:\some\path" it needs to be "c:/some/path")

If you don't have an Android SDK you can go back to using the original APK generation mechanism which means you'd need to edit your app descriptor file to add a `<BuildLegacyAPK>true</BuildLegacyAPK>` section under the `<android>` part..

thanks

**Comment by marcanw** [comment]
Hi @ajwfrost It works great now.
Can I use the build 698 for the release or do I have to wait the official AIR Sdk? 
Thank you
